redfish-core/lib/FabricAdapters.hpp

```diff
@@ -13,6 +14,130 @@ using MapperGetSubTreeResponse = std::vector<
+ * @param[in]       objPath     Adapter Dbus object path.
+ * @param[in]       serviceMap  Map to hold service and interface.
+ */
+inline void
+    getAdapterProperties(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                         const std::string& objPath,
+                         const ServiceMap& serviceMap)
+{
+    name_util::getPrettyName(aResp, objPath, serviceMap, "/Name"_json_pointer);
+
+    for (const auto& [serviceName, interfaceList] : serviceMap)
+    {
+        for (const auto& interface : interfaceList)
+        {
+            if (interface == "xyz.openbmc_project.Inventory.Decorator.Asset")
+            {
+                crow::connections::systemBus->async_method_call(
+                    [aResp, objPath](const boost::system::error_code ec,
+                                     const PropertyListType& propertiesList) {
+                        if (ec)
+                        {
+                            BMCWEB_LOG_DEBUG << "DBUS response error";
+                            messages::internalError(aResp->res);
+                            return;
+                        }
+
+                        for (const PropertyType& property : propertiesList)
+                        {
+                            if (property.first == "PartNumber")
+                            {
+                                const std::string* value =
+                                    std::get_if<std::string>(&property.second);
+                                if (value == nullptr)
+                                {
+                                    messages::internalError(aResp->res);
+                                    return;
+                                }
+                                aResp->res.jsonValue["PartNumber"] = *value;
+                            }
+                            else if (property.first == "SerialNumber")
+                            {
+                                const std::string* value =
+                                    std::get_if<std::string>(&property.second);
+                                if (value == nullptr)
+                                {
+                                    messages::internalError(aResp->res);
+                                    return;
+                                }
+                                aResp->res.jsonValue["SerialNumber"] = *value;
+                            }
+                            else if (property.first == "SparePartNumber")
+                            {
+                                const std::string* value =
+                                    std::get_if<std::string>(&property.second);
+                                if (value == nullptr)
+                                {
+                                    messages::internalError(aResp->res);
+                                    return;
+                                }
+                                aResp->res.jsonValue["SparePartNumber"] =
```

Comment:
Giving the same comment Ed gave upstream for another commit: Since SparePartNumber has a default of "", please omit adding it if the value within is a blank to avoid warnings.